src/main/java/com/project/controller/AuthenticationController.java

```diff
@@ -0,0 +1,57 @@
+package com.project.controller;
+
+import com.project.model.AuthenticationRequest;
+import com.project.repository.UserRepository;
+import com.project.security.JwtTokenProvider;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.springframework.http.ResponseEntity.ok;
+
+@RestController
+@RequestMapping("/auth")
+public class AuthenticationController {
+
+    private final AuthenticationManager authenticationManager;
+    private final JwtTokenProvider jwtTokenProvider;
+    private final UserRepository users;
+
+    @Autowired
+    public AuthenticationController(AuthenticationManager authenticationManager, JwtTokenProvider jwtTokenProvider, UserRepository users) {
+        this.authenticationManager = authenticationManager;
+        this.jwtTokenProvider = jwtTokenProvider;
+        this.users = users;
+    }
+
+    @PostMapping("/login")
+    public ResponseEntity<Map<Object, Object>> login(@RequestBody AuthenticationRequest data) {
```

Comment:
I suggest creating something like `AuthenticationResponse` and returning it. Also, I don't think you need to return the user's email. The token should be sufficient enough unless there is some kind of requirement for it.

Reply:
created response entities for ApplicationRequest, AuthenticationRequest, CommentRequest, ApplicationResponse, AuthenticationResponse, CommentResponse and UserResponse to reduce the ammount of information moving through controllers between front end and back end